### PR TITLE
[SPARK-43662][PS][CONNECT] Support merge_asof in Spark Connect

### PR DESCRIPTION
**Pull request by ueshin**
### What changes were proposed in this pull request?

Supports `merge_asof` in Spark Connect.

### Why are the changes needed?

`merge_asof` is missing in Spark Connect.

Ref: https://spark.apache.org/docs/latest/api/python/reference/pyspark.pandas/api/pyspark.pandas.merge_asof.html

### Does this PR introduce _any_ user-facing change?

Yes, `merge_asof` is available in Spark Connect.

### How was this patch tested?

The parity tests.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by ueshin** [comment]
cc @itholic @HyukjinKwon @zhengruifeng

**Review by HyukjinKwon — APPROVED** [review]

**Review by zhengruifeng — APPROVED** [review]

File: python/pyspark/sql/connect/dataframe.py [hunk 1]
```diff
@@ -595,6 +595,47 @@ def join(
 
     join.__doc__ = PySparkDataFrame.join.__doc__
 
+    def _joinAsOf(
```
Comment (zhengruifeng) [inline python/pyspark/sql/connect/dataframe.py:598]: I have a dumb question: why this method name start with `_`?

Comment (ueshin) [reply python/pyspark/sql/connect/dataframe.py:598]: Because so far this is supposed to be called from pandas API's `merge_asof` only.

**Review by itholic — APPROVED** [review]

**Comment by zhengruifeng** [comment]
merged to master

**Pull request closed without merging**